### PR TITLE
Moved from the old osg-flock scripts to the current GWMS singularity … (SOFTWARE-4725)

### PR DESCRIPTION
**Pull request by rynge**
…ones

### CODE REVIEW

File: 10-setup-htcondor.sh [hunk 1]
```diff
@@ -133,3 +217,51 @@ echo
...
+
+# set some defaults for the glideinwms based scripts
+cat >$glidein_config <<EOF
+ADD_CONFIG_LINE_SOURCE $PWD/add_config_line.source
+ALLOW_NONCVMFS_IMAGES True
+CONDOR_VARS_FILE $condor_vars_file
+ERROR_GEN_PATH $PWD/error_gen.sh
+GLIDEIN_SINGULARITY_REQUIRE OPTIONAL
+GLIDEIN_Singularity_Use PREFERRED
+OSG_DEFAULT_CONTAINER_DISTRIBUTION 70%__opensciencegrid/osgvo-el7:latest 30%__opensciencegrid/osgvo-el8:latest
```
Comment (dsschult) [inline 10-setup-htcondor.sh:243]: I assume this is setting the default singularity container, which will get used if singularity is available and the job does not specify an image.

Is there a way to make this configurable?

Comment (rynge) [reply 10-setup-htcondor.sh:243]: We could expose it as another env var. Or do you have another suggestion on how to interface with container configuration?

Comment (dsschult) [reply 10-setup-htcondor.sh:243]: Yes, env vars work for me.  That would allow me to pick el7 100% of the time.

Comment (rynge) [reply 10-setup-htcondor.sh]: Ok, it is now picked up as an env var. You should be able to do:

``
export OSG_DEFAULT_CONTAINER_DISTRIBUTION="100%__opensciencegrid/osgvo-el7:latest"
```

The image part can be anything available under `/cvmfs/singularity.opensciencegrid.org/`

File: 10-setup-htcondor.sh [hunk 2]
```diff
@@ -7,6 +7,87 @@ if [ `id -u` = 0 ]; then
     exit 1
 fi
 
+
+#
+# Taken from creation/web_base/condor_startup.sh in the gwms
```
Comment (matyasselmeci) [inline 10-setup-htcondor.sh:12]: I guess we have to do this for now but we should try and get upstream to move this function to a sourceable `condor_lib.sh` file, like how they have `singularity_lib.sh`.

File: 10-setup-htcondor.sh [hunk 3]
```diff
@@ -133,3 +217,54 @@ echo
...
+export glidein_config=$LOCAL_DIR/glidein_config
+export condor_vars_file=$LOCAL_DIR/main/condor_vars.lst
+
+# set some defaults for the glideinwms based scripts
+if [[ -z $OSG_DEFAULT_CONTAINER_DISTRIBUTION ]]; then
+    OSG_DEFAULT_CONTAINER_DISTRIBUTION="70%__opensciencegrid/osgvo-el7:latest 30%__opensciencegrid/osgvo-el8:latest"
+fi
+cat >$glidein_config <<EOF
+ADD_CONFIG_LINE_SOURCE $PWD/add_config_line.source
+ALLOW_NONCVMFS_IMAGES True
```
Comment (matyasselmeci) [inline 10-setup-htcondor.sh:241]: This one **must** be configurable, and off by default, at least until we document it. As we're seeing, non-CVMFS images use a lot of disk and network, and the site admin has no way of restricting this.

Comment (rynge) [reply 10-setup-htcondor.sh:241]: Yeah, that is probably a mistake - let me take another test pass

Comment (rynge) [reply 10-setup-htcondor.sh]: Fixed. I also added the latest stashcp, as that is an important component if /cvmfs it not available.

File: 10-setup-htcondor.sh [hunk 4]
```diff
@@ -133,3 +217,60 @@ echo
...
+OSG_DEFAULT_CONTAINER_DISTRIBUTION $OSG_DEFAULT_CONTAINER_DISTRIBUTION
+SINGULARITY_IMAGE_RESTRICTIONS None
+EOF
+touch $condor_vars_file
+
+# grab the latests copy of stashcp
+mkdir -p client
+
+if (curl --silent --fail --location --connect-timeout 30 --speed-limit 1024 -o client/stashcp http://stash.osgconnect.net/public/dweitzel/stashcp/current/stashcp) &>/dev/null; then
+    chmod 755 client/stashcp
```
Comment (matyasselmeci) [inline 10-setup-htcondor.sh:254]: What puts "client" in the PATH?

Comment (rynge) [reply 10-setup-htcondor.sh:254]: That is part of the expected gwms directory structure. We just have to mimic it here to make it all work together.

**Review by matyasselmeci — APPROVED** [review]
OK, let's do this!

**Pull request merged**